# Testing with thrown errors

For all these exercises copy your existing code to a new folder named `thrown-error-test`. Then update your code with the given instructions.

## Exercise 1

Update [Problem 1 from the synchronous testing exercises](synchronous-testing.md) to throw/catch errors if the is a division by zero

## Exercise 2

Update [Problem 2 from the synchronous testing exercises](synchronous-testing.md) to throw/catch errors if the input is below zero

## Exercise 3

Update [Problem 4 from the synchronous testing exercises](synchronous-testing.md) so that all 4 validation criteria can be set and are process when the function is called. Also, have the following conditions throw errors:

- If the maximum is set below minimum, throw error;
- If minimum is set above maximum, throw error;
- If when setting the invalid value array, it shares values with the valid value array, throw error.
- If when setting the valid value array, it shares valies with the invalid value array, throw error.

## Exercise 4

Update [Problem 1 from the asynchronous exercises](asynchronous-testing.md) to throw/catch errors if there are any problems with the file access.